fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java

```diff
@@ -563,7 +558,20 @@ private void addScanRangeLocations(Partition partition,
 
             result.add(scanRangeLocations);
         }
+
         // FIXME(dhc): we use cardinality here to simulate ndv
+        // update statsDeriveResult for real statistics
+        // After statistics collection is complete, remove the logic
+        if (analyzer.safeIsEnableJoinReorderBasedCost()) {
+            statsDeriveResult.setRowCount(cardinality);
+            for (Map.Entry<Long, Long> entry : statsDeriveResult.getColumnToNdv().entrySet()) {
+                if (entry.getValue() > 0) {
+                    cardinality = Math.min(cardinality, entry.getValue());
```

Comment:
I understand you are here to re-update statsresult.
But actually I think this logic should be more reasonable in computestats.

Reply:
done